scheduler/src/grpc/agent_workload_service.rs

```diff
@@ -0,0 +1,20 @@
+//! Workload gRPC service for the Orka node agents.
+
+use orka_proto::node_agent::workload_service_client::WorkloadServiceClient;
+use orka_proto::node_agent::Workload;
+use tonic::Request;
+
+#[tokio::main]
+pub async fn create(workload: Workload) -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
As explained further in the comments regarding the `schedule` function, this function should return a `Result<Streaming<WorkloadStatus>, Status>`. This enables better and clearer communication, whilst also forwarding errors that can be nicely caught and raised outside of this function.

I will let you implement the function to match the signature.

```suggestion
pub async fn create(workload: Workload) -> Result<Streaming<WorkloadStatus>, Status> {
```

Reply:
Of course !